unittests/test_utility.py

```diff
@@ -30,6 +30,15 @@ def test_command_timeout(self):
         except SpawnedProcessTimeout as e:
             self.assertEqual(e.timeout, 2)
 
+    def test_command_timeout_exception_string(self):
+        with self.assertRaises(SpawnedProcessTimeout) as e:
+            os_ext.run_command('sleep 3', timeout=2)
+
+        try:
+            str(e.exception)
+        except TypeError:
+            self.fail('could not convert exception to string')
```

Comment:
There is no need for a new test function here. Simply try to convert the exception thrown in the previous test. Also, you don't need to catch anything. If any exception is thrown, the unit tests will fail in any way. So the following is enough:

```python
# Try to get the string representation of the exception: see bug #XXX 
s = str(e)
```
I don't understand as well the `e.exception`?

Reply:
`e` is the context. To take the thrown exception you use the `e.exception` but since I will add to the previous test there is no need for it.